lib/purescript-ast/src/Language/PureScript/Names.hs

```diff
@@ -241,7 +236,14 @@ isQualifiedWith _ _ = False
 
 $(deriveJSON (defaultOptions { sumEncoding = ObjectWithSingleField }) ''Qualified)
 $(deriveJSON (defaultOptions { sumEncoding = ObjectWithSingleField }) ''Ident)
-$(deriveJSON (defaultOptions { sumEncoding = ObjectWithSingleField }) ''ModuleName)
+
+instance ToJSON ModuleName where
```

Comment:
I'm continuing to encode module names as arrays of strings in JSON to not break the compatibility tests. I'm assuming changing this would break Pursuit. Would Pursuit be fine if my handwritten instance accepted both a single string as well as an array of them?